backpack/core/derivatives/avgpoolnd.py

```diff
@@ -0,0 +1,179 @@
+            out_features = C * H_out * W_out
+            shape_out = (1, H_out, W_out)
+        elif self.N == 3:
+            _, C, D_in, H_in, W_in = module.input0.size()
+            _, _, D_out, H_out, W_out = module.output.size()
+            in_features = C * D_in * H_in * W_in
+            out_features = C * D_out * H_out * W_out
+            shape_out = (1, D_out, H_out, W_out)
+        else:
+            raise ValueError(
+                "{}-dimensional Maxpool. is not implemented.".format(self.N)
+            )
+
+        mat = mat.reshape(out_features * C, *shape_out)
+        jac_t_mat = self.__apply_jacobian_t_of(module, mat).reshape(
+            out_features, in_features
+        )
+        mat_t_jac = jac_t_mat.t().reshape(in_features * C, *shape_out)
+        jac_t_mat_t_jac = self.__apply_jacobian_t_of(module, mat_t_jac).reshape(
+            in_features, in_features
+        )
+
+        return jac_t_mat_t_jac.t()
+
+    def check_exotic_parameters(self, module):
+        assert module.count_include_pad, (
+            "Might not work for exotic hyperparameters of AvgPool2d, "
+            + "like count_include_pad=False"
+        )
+
+    def _jac_mat_prod(self, module, g_inp, g_out, mat):
+        self.check_exotic_parameters(module)
+
+        mat_as_pool = self.__make_single_channel(mat, module)
+        jmp_as_pool = self.__apply_jacobian_of(module, mat_as_pool)
+        self.__check_jmp_out_as_pool(mat, jmp_as_pool, module)
+
+        return self.reshape_like_output(jmp_as_pool, module)
+
+    def __make_single_channel(self, mat, module):
+        """Create fake single-channel images, grouping batch,
+        class and channel dimension."""
+        result = eingroup(self.equation, mat)
+        C_axis = 1
+        return result.unsqueeze(C_axis)
+
+    def __apply_jacobian_of(self, module, mat):
+        convnd = self.conv(
+            in_channels=1,
+            out_channels=1,
+            kernel_size=module.kernel_size,
+            stride=module.stride,
+            padding=module.padding,
+            bias=False,
+        ).to(module.input0.device)
+
+        convnd.weight.requires_grad = False
+        avg_kernel = torch.ones_like(convnd.weight) / convnd.weight.numel()
+        convnd.weight.data = avg_kernel
+
+        return convnd(mat)
+
+    def __check_jmp_out_as_pool(self, mat, jmp_as_pool, module):
+        V = mat.size(0)
+        if self.N == 1:
+            N, C_out, L_out = module.output.shape
+            assert jmp_as_pool.shape == (V * N * C_out, 1, L_out)
+        elif self.N == 2:
+            N, C_out, H_out, W_out = module.output.shape
+            assert jmp_as_pool.shape == (V * N * C_out, 1, H_out, W_out)
+        elif self.N == 3:
+            N, C_out, D_out, H_out, W_out = module.output.shape
+            assert jmp_as_pool.shape == (V * N * C_out, 1, D_out, H_out, W_out)
+        else:
+            raise ValueError("{}-dimensional Conv. is not implemented.".format(self.N))
```

Comment:
I think we can drop that statement, since the proper range of `self.N` is already checked in the constructor